lib/aptly/app.rb

```diff
@@ -2,11 +2,48 @@ module Aptly
   class App < Resource
     def_attr :id
     def_attr :handle
+    def_attr :operations_href, "_links", "operations", "href"
+    def_attr :current_configuration_href, "_links", "current_configuration", "href"
 
     def services
       @data.dig("_embedded", "services").map do |service_data|
         Service.new(@access_token, service_data)
       end
     end
+
+    # The Operation returned by #deprovision! may return a 404 error
+    # when reloaded or when #wait_for_completion is called on it,
+    # because once the app is destroyed, the operation URL doesn't
+    # refer to anything. So callers should be ready to deal with 404s.
+    def deprovision!
+      response = @access_token.post(
+        operations_href,
+        body: { type: "deprovision" }.to_json,
+        headers: { "Content-Type" => "application/json" }
+      )
+      Operation.new(@access_token, response.parsed)
+    end
+
+    def current_configuration
+      response = @access_token.get(current_configuration_href)
```

Comment:
such a nice opportunity to use:

```ruby
@access_token.get(current_configuration_href).then do |response|
  Configuration.new(@access_token, response.parsed)
end
```

:P

Reply:
Yeah, but this lib accepts rubies >= 2.5.

Reply:
oh :(